modules/search/class.jetpack-search-widget-filters.php

```diff
@@ -140,10 +149,94 @@ function widget( $args, $instance ) {
 			do_action( 'jetpack_search_render_filters_widget_title', $title, $args['before_title'], $args['after_title'] );
 		}
 
+		// we need to dynamically inject the sort field into the search box when the search box is enabled, and display
+		// it separately when it's not.
 		if ( ! empty( $instance['search_box_enabled'] ) ) {
 			$this->render_widget_search_form( $instance );
 		}
 
+		$default_sort = isset( $instance['sort'] ) ? $instance['sort'] : self::DEFAULT_SORT;
+		list( $orderby, $order ) = $this->sorting_to_wp_query_param( $default_sort );
+		$current_sort = "{$orderby}|{$order}";
+
+		if ( ! empty( $instance['search_box_enabled'] ) && ! empty( $instance['user_sort_enabled'] ) ) {
+			?>
+			<label class="jetpack-search-sort-wrapper">
+				<?php esc_html_e('Sort by', 'jetpack'); ?>
+				<select name="<?php echo esc_attr( $this->get_field_name( 'sort' ) ); ?>" class="jetpack-search-sort">
+					<?php foreach( $this->get_sort_types() as $sort => $label ) { ?>
+						<option value="<?php echo $sort; ?>" <?php selected( $current_sort, $sort ); ?>>
+							<?php echo $label; ?>
+						</option>
+					<?php } ?>
+				</select>
+			</label> <?php
+		}
+
+		/*
+		 * this JS is a bit complicated, but here's what it's trying to do:
+		 * - find or create a search form
+		 * - find or create the orderby/order fields with default values
+		 * - detect changes to the sort field, if it exists, and use it to set the order field values
+		 */
+		?>
+		<script type="text/javascript">
```

Comment:
We should probably enqueue jQuery on the frontend if we're going to drop this in.

Reply:
This JavaScript gets added to the page even if "Show sorting controls" is not toggled.

Reply:
In the current implementation the JS is still necessary since it also add the hidden sorting fields to the form with the default sort values.

We can move it into the `if` statement once we move to injecting the hidden fields using PHP rather than javascript.

Reply:
(ie. in a separate patch)